Allocation UI page is missing information when browsed from Job UI page

### Nomad version
v1.3.6

### Operating system and Environment details

* Chrome 113.0.5672.126 (and earlier)

### Issue

When navigating to an allocation's page in the UI from the job's page via the "Recent Allocations" link, the allocation's page is missing information.

Refreshing the page or opening the allocation page's URL in a new tab fixes the issue.

### Reproduction steps

1. For a job that has at least one running alloc, go to its page (e.g. `https://.../ui/jobs/JobName`)
2. Click on one of the running allocs in the Recent Allocations table

#### Expected Result

* The CPU/Memory graphs are non-empty (i.e. show something other than empty graphs & `0 ... / 0 ... Total`)
* The Task Lifecycle Status table is non-empty
* The Tasks/Ports/Services tables appear

#### Actual Result

See screenshot:

<img width="1428" alt="nomad-gh-bugreport-ui" src="https://github.com/hashicorp/nomad/assets/48605613/9dbb4d0d-76d5-4a2a-8797-3f7997b2ed89">

Hi @philrenaud, I appreciate you looking into it.

> What's the job type being run here (service? system?)

This is primarily for service jobs; we haven't had reports of this in system jobs, but we also have comparatively much fewer of those, so there are fewer interactions that would've led to a report

> Are you logged in with particular ACL policy/role rules applied? If so, could you let me know what your permissions look like?

No, we are not logged in with any ACL tokens when this happens. No secret or accessor ID is set in the UI when this happens

> Are you seeing any javascript errors on the page? I notice in your screenshot that the keyboard shortcuts (light green letter boxes) are squished to the top left corner of your screen; makes me think that a JS error has taken place before these could be positioned. If there is an error, could you please provide some details for it?

I'll be sure to check the next time this happens and report back with any inspector errors.

What I can say is that there definitely are some console errors (e.g. a 500 Internal Server error consistently appears on page load against path `/v1/acl/token/self`, and 100+ `chunk.514.007cad389153bda00b8f.js:2412 Error: <rect> attribute width: A negative value is not valid. ("-...px")` error logs are in the console) but **without** this issue manifesting. I'm not sure if this rules anything out, but hopefully it's useful troubleshooting data.

--- a/ui/app/controllers/clients/client/index.js
+++ b/ui/app/controllers/clients/client/index.js
@@ -205,7 +205,7 @@ export default class ClientController extends Controller.extend(
 
   @action
   gotoAllocation(allocation) {
-    this.transitionToRoute('allocations.allocation', allocation);
+    this.transitionToRoute('allocations.allocation', allocation.id);
   }
 
   @action

--- a/ui/app/controllers/csi/plugins/plugin/allocations.js
+++ b/ui/app/controllers/csi/plugins/plugin/allocations.js
@@ -113,7 +113,7 @@ export default class AllocationsController extends Controller.extend(
   @action
   gotoAllocation(allocation, event) {
     lazyClick([
-      () => this.transitionToRoute('allocations.allocation', allocation),
+      () => this.transitionToRoute('allocations.allocation', allocation.id),
       event,
     ]);
   }

--- a/ui/app/controllers/csi/plugins/plugin/index.js
+++ b/ui/app/controllers/csi/plugins/plugin/index.js
@@ -19,6 +19,6 @@ export default class IndexController extends Controller {
 
   @action
   gotoAllocation(allocation) {
-    this.transitionToRoute('allocations.allocation', allocation);
+    this.transitionToRoute('allocations.allocation', allocation.id);
   }
 }

--- a/ui/app/controllers/csi/volumes/volume.js
+++ b/ui/app/controllers/csi/volumes/volume.js
@@ -60,6 +60,6 @@ export default class VolumeController extends Controller {
 
   @action
   gotoAllocation(allocation) {
-    this.transitionToRoute('allocations.allocation', allocation);
+    this.transitionToRoute('allocations.allocation', allocation.id);
   }
 }

--- a/ui/app/controllers/jobs/job/allocations.js
+++ b/ui/app/controllers/jobs/job/allocations.js
@@ -166,7 +166,7 @@ export default class AllocationsController extends Controller.extend(
 
   @action
   gotoAllocation(allocation) {
-    this.transitionToRoute('allocations.allocation', allocation);
+    this.transitionToRoute('allocations.allocation', allocation.id);
   }
 
   get optionsAllocationStatus() {

--- a/ui/app/controllers/jobs/job/task-group.js
+++ b/ui/app/controllers/jobs/job/task-group.js
@@ -140,7 +140,7 @@ export default class TaskGroupController extends Controller.extend(
 
   @action
   gotoAllocation(allocation) {
-    this.transitionToRoute('allocations.allocation', allocation);
+    this.transitionToRoute('allocations.allocation', allocation.id);
   }
 
   @action

--- a/ui/app/routes/allocations/allocation.js
+++ b/ui/app/routes/allocations/allocation.js
@@ -50,6 +50,9 @@ export default class AllocationRoute extends Route.extend(WithWatchers) {
         super.model(...arguments),
         this.store.findAll('namespace'),
       ]);
+      if (allocation.isPartial) {
+        await allocation.reload();
+      }
       const jobId = allocation.belongsTo('job').id();
       await this.store.findRecord('job', jobId);
       return allocation;

--- a/ui/tests/acceptance/job-allocations-test.js
+++ b/ui/tests/acceptance/job-allocations-test.js
@@ -4,7 +4,7 @@
  */
 
 /* eslint-disable qunit/require-expect */
-import { currentURL } from '@ember/test-helpers';
+import { currentURL, click, find } from '@ember/test-helpers';
 import { module, test } from 'qunit';
 import { setupApplicationTest } from 'ember-qunit';
 import { setupMirage } from 'ember-cli-mirage/test-support';
@@ -75,6 +75,33 @@ module('Acceptance | job allocations', function (hooks) {
     });
 
     assert.equal(document.title, `Job ${job.name} allocations - Nomad`);
+  });
+
+  test('clicking an allocation results in the correct endpoint being hit', async function (assert) {
+    server.createList('allocation', Allocations.pageSize - 1, {
+      shallow: true,
+    });
+    allocations = server.schema.allocations.where({ jobId: job.id }).models;
+
+    await Allocations.visit({ id: job.id });
+
+    const firstAllocation = find('[data-test-allocation]');
+    await click(firstAllocation);
+
+    const requestToAllocationEndpoint = server.pretender.handledRequests.find(
+      (request) =>
+        request.url.includes(
+          `/v1/allocation/${firstAllocation.dataset.testAllocation}`
+        )
+    );
+
+    assert.ok(requestToAllocationEndpoint, 'the correct endpoint is hit');
+
+    assert.equal(
+      currentURL(),
+      `/allocations/${firstAllocation.dataset.testAllocation}`,
+      'the URL is correct'
+    );
   });
 
   test('allocations table is sortable', async function (assert) {

